SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -439,24 +440,26 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit> > const
     std::vector<PCaloHit> hcalHits;
     hcalHits.reserve(hcalHitsOrig.size());
 
-    for ( unsigned int i=0; i< hcalHitsOrig.size(); i++) {
-      DetId id(hcalHitsOrig[i].id());
-      HcalDetId hid(id);
-      
-      //      if ( id.subdetId()==2 && hid.ietaAbs()<16 ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      if ( !htopoP->validHcal(hid) ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      else
-	hcalHits.push_back(hcalHitsOrig[i]);
-    }
     //evaluate darkening before relabeling
     if(m_HEDarkening || m_HFRecalibration){
-      darkening(hcalHits);
+      darkening(hcalHitsOrig);
```

Comment:
@davidlange6 In phase 2 we use hcaltestnumbering scheme which relabels the DetId's and the content gets changed - so the validity test has too come after the relabelling.
________________________________
From: Lindsey Gray [notifications@github.com]
Sent: 05 November 2015 15:51
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] HGC ToT Digitization Implementation (and corresponding RECO updates) (#11880)


In SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc<https://github.com/cms-sw/cmssw/pull/11880#discussion_r44019372>:

>      //evaluate darkening before relabeling
>      if(m_HEDarkening || m_HFRecalibration){
> -      darkening(hcalHits);
> +      darkening(hcalHitsOrig);


@bsunanda<https://github.com/bsunanda> Please answer this.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/11880/files#r44019372>.